tests/Integration/Behaviour/Features/Context/Domain/Discount/DiscountFeatureContext.php

```diff
@@ -188,6 +189,61 @@ public function createDiscount(string $discountReference, string $discountType,
         }
     }
 
+    /**
+     * @When I update discount :discountType with the following properties:
+     */
+    public function editDiscount(string $discountReference, TableNode $node): void
+    {
+        $data = $this->localizeByRows($node);
+        $discountId = $this->getSharedStorage()->get($discountReference);
+        $command = new UpdateDiscountCommand($discountId);
+
+        if (isset($data['name'])) {
+            $command->setLocalizedNames($data['name']);
+        }
+        if (isset($data['highlight'])) {
+            $command->setHighlightInCart(PrimitiveUtils::castStringBooleanIntoBoolean($data['highlight']));
+        }
+        if (isset($data['allow_partial_use'])) {
+            $command->setAllowPartialUse(PrimitiveUtils::castStringBooleanIntoBoolean($data['allow_partial_use']));
+        }
+        if (isset($data['priority'])) {
+            $command->setPriority((int) $data['priority']);
+        }
+        if (isset($data['active'])) {
+            $command->setActive(PrimitiveUtils::castStringBooleanIntoBoolean($data['active']));
+        }
+        if (isset($data['valid_from'])) {
+            if (empty($data['valid_to'])) {
+                throw new RuntimeException('When setting cart rule range "valid_from" and "valid_to" must be provided');
+            }
+            $command->setValidityDateRange(
+                new DateTimeImmutable($data['valid_from']),
+                new DateTimeImmutable($data['valid_to']),
+            );
+        }
+        if (isset($data['total_quantity'])) {
+            $command->setTotalQuantity((int) $data['total_quantity']);
+        }
+
+        if (isset($data['quantity_per_user'])) {
+            $command->setQuantityPerUser((int) $data['quantity_per_user']);
+        }
+
+        $command->setDescription($data['description'] ?? '');
+        if (!empty($data['code'])) {
+            $command->setCode($data['code']);
+        }
+
+        try {
+            /** @var DiscountId $discountId */
+            $discountId = $this->getCommandBus()->handle($command);
+            $this->getSharedStorage()->set($discountReference, $discountId->getValue());
```

Comment:
```suggestion
            $this->getCommandBus()->handle($command);
```
You don't need to updated the ID since it's never supposed to change (see comment above in the handler)